src/opentimelineio/serializableObject.h

```diff
@@ -543,5 +547,11 @@ class SerializableObject {
     AnyDictionary _dynamic_fields;
     friend class TypeRegistry;
 };
+
+template <class T, class U>
+SerializableObject::Retainer<T> dynamic_retainer_cast(SerializableObject::Retainer<U> const& retainer)
```

Comment:
I wonder if we need a different name for this? given class Foo, and class Bar::Foo, std:::dynamic_ptr_cast<Bar*>(myFoo) will return a shared_ptr<Bar>. dynamic_retainer_cast actually does a value unboxing, so think that perhaps there is some other C++ idiom we need here. https://en.cppreference.com/w/cpp/types/remove_reference ~ might remove_retainer make more sense?

Reply:
dynamic_value_cast maybe? Since remove_reference doesn't also do a cast?

Reply:
The dynamic_retainer_cast returns a Retainer by automatically constructing it from the unboxed value, isn't that similar to std:::dynamic_ptr_cast?

Reply:
Ah right. Gremlins munching on my cerebrum, I focused on the return T, not the implicit conversion. I am now recalling we talked this through aeons ago ;)

Reply:
No worries! I forgot, did we decide if we wanted the other casts as well, i.e. static?